### PR TITLE
feat: pass default adapter to adapter option

### LINKED ISSUE #181: Interceptors?
Hey friends, I found this module via https://github.com/JustinBeckwith/retry-axios/issues/1, and tried it out as a 'drop-in replacement' for axios because I need that sweet retry goodness and `retry-axios` has this issue where it only retries once, then seems to...forget my config. Something seems to be missing, though: interceptors!

**Is your feature request related to a problem? Please describe.**
I think this is a feature request, unless I'm _really bad_ at reading documentation. Couldn't see an indication in the source code that it's supported either.

So I copy/paste replaced `axios` with `gaxios`, ran my code...

```javascript
const requests = require('gaxios'),
  _ = require('lodash')

const logHttpErrorDetails = requests.interceptors.response.use(/* my fully sick function for logging all error responses */)
```

...and got this error:
```TypeError: Cannot read property 'response' of undefined```

 **Describe the solution you'd like**
I would like axios-like interceptors supported for use in `gaxios`.

 **Describe alternatives you've considered**
Probably just going back to `axios` and putting up with my `retry-axios` issue.

 **Additional context**
🙏

### PR DESCRIPTION
**Pull request by elierotenberg**
* enable request-level instrumentation (e.g. timings, tracing...) as
  discussed in #181

* add chai as a devDependency and fix webpack config to pass tests

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com//issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #181 🦕

### CODE REVIEW

File: browser-test/test.browser.ts [hunk 1]
```diff
@@ -14,15 +14,9 @@
 import assert from 'assert';
 import {describe, it} from 'mocha';
 import {request} from '../src/index';
-import {PassThrough} from 'stream';
-import * as stream from 'stream';
 import * as uuid from 'uuid';
 const port = 7172; // should match the port defined in `webserver.ts`
 
-function isReadableStream(obj: stream.Readable | string) {
```
Comment (bcoe) [inline browser-test/test.browser.ts:22]: was this just not used?

**Review by bcoe — APPROVED** [review]
I really like this feature 👍 thank you for the contribution.

File: browser-test/test.browser.ts (hunk 1 above)
Comment (elierotenberg) [reply browser-test/test.browser.ts:22]: Yes, apparently! TypeScript raised a warning and there were no errors in tests after I deleted it so I assumed it was okay.

**Comment by bcoe** [comment]
@elierotenberg your work is now released in `3.1.0`, please try it out and let us know if it works as expected.

**Pull request merged**